src/public/modules/forms/base/directives/field.client.directive.js

```diff
@@ -23,6 +28,27 @@ function fieldDirective(FormFields) {
       isValidateDate: '<',
     },
     link: function (scope) {
+      // Stealth prefill feature
+      // If a query parameter is provided to a form URL in the form ?<fieldId1>=<value1>&<fieldId2>=<value2>...
+      // And if the fieldIds are valid mongoose object IDs and refer to a short text field,
+      // Then prefill and disable editing the corresponding form field on the frontend
+
+      const queryParams = $location.search()
+      console.log(queryParams)
+
+      if (
+        scope.field._id in queryParams &&
+        scope.field.fieldType === 'textfield'
+      ) {
+        let prefillValue = queryParams[scope.field._id]
+        prefillValue = Array.isArray(prefillValue)
+          ? prefillValue[prefillValue.length - 1] // use last value in array if param key repeated
+          : // but note that there is no specification that the last value will definitely be last in order in the url string
+            prefillValue
+        scope.field.fieldValue = $sanitize(prefillValue) // sanitize prefillValue as a precaution, but this means "<>& cannot be used
+        scope.field.disabled = true
```

Comment:
I thought the case was to prefill but leave it modifiable? What was the conclusion? @liangyuanruo @syan-syan

Reply:
@syan-syan @karrui based on what liwei told me (intent is for public to scan the form to prefill a serial number) I think it should be non-modifiable, since the purpose is to avoid manual entry errors

Reply:
that is correct